src/models/User.js

```diff
@@ -0,0 +1,37 @@
+const mongoose = require('mongoose');
+
+const UserSchema = new mongoose.Schema({
+  firstName: {
+    type: String,
+    required: [true, 'Please provide a first name'],
+    minlength: [2, 'First name must be at least 2 characters'],
+    maxlength: [50, 'First name must be at most 50 characters'],
+  },
+  lastName: {
+    type: String,
+    required: [true, 'Please provide a last name'],
+    minlength: [2, 'Last name must be at least 2 characters'],
+    maxlength: [50, 'Last name must be at most 50 characters'],
+  },
+  email: {
+    type: String,
+    required: [true, 'Please provide an email'],
+    unique: true,
+    match: [
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+      'Please provide a valid email',
+    ],
+  },
+  password: {
+    type: String,
+    required: [true, 'Please provide password'],
+    minlength: 6,
+  },
+  role: {
```

Comment:
If I'm not mistaken we decided not to specify any role. Every user can do both sell and purchase.

Reply:
I will add location then and the seller's books we'll add when the booksSchema will be ready